src/cpp/utils/SystemInfo.cpp

```diff
@@ -83,4 +86,35 @@ bool SystemInfo::file_exists(
     return (stat(filename.c_str(), &s) == 0 && s.st_mode & S_IFREG);
 }
 
+ReturnCode_t SystemInfo::load_environment_file(
+        const std::string& filename,
+        const std::string& env_name,
+        std::string& env_value)
+{
+    // Check that the file exists
+    if (!SystemInfo::file_exists(filename))
+    {
+        return ReturnCode_t::RETCODE_BAD_PARAMETER;
+    }
+
+    // Read json file
+    std::ifstream file(filename);
+    nlohmann::json file_content;
+    file >> file_content;
```

Comment:
What if the file is empty or the json is malformed? This will throw. We should return some kind of error code, probably RETCODE_ERROR.

Reply:
addressed in f67682c

Reply:
I have completed the doxygen documentation to include this new case in afb20af49b76a0b9f1eb3b9241f468218c87b0b0